### PR TITLE
Improve HelperOptions doc

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.x 
| Description?  | Improve HelperOptions doc: more examples and new array notation
| Fixed ticket? | old pr here https://github.com/PrestaShop/docs/pull/1075

### CODE REVIEW

File: development/components/helpers/helperoptions.md [hunk 1]
```diff
@@ -6,50 +6,57 @@ title: HelperOptions
...
-                                                                   // or strval depending on value type you want to receive.
-        'type' => {'text', 'hidden', 'select', 'bool', 'radio',    // The kind of input field you want to use.
+$this->fields_options = [
+  'general' => [
+    'title' => $this->l('Carrier options'),                      // The title of the fieldset. If missing, default is 'Options'.
+    'top' => $this->l('Text to display before the fieldset'),    // This text is display right above the first. Rarely used.
+    'image' => 'url to icon',                                    // If missing, will use the default icon for the tab.
+    'description' => $this->l('Display as description'),         // Displays an informational box above the fields.
+    'info' => $this->l('Display as info'),                       // Displays an unstyled text above the fields.
+    'fields' => array(                                           // The various option fields.
```
Comment (Progi1984) [inline development/components/helpers/helperoptions.md]:
```suggestion
    'fields' => [                                                // The various option fields.
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

File: development/components/helpers/helperoptions.md (hunk 1 above)
Comment (jf-viguier) [reply development/components/helpers/helperoptions.md]: done @Progi1984

**Comment by jf-viguier** [comment]
@Progi1984 is it ok now ?

**Review by Progi1984 — APPROVED** [review]

File: development/components/helpers/helperoptions.md [hunk 2]
```diff
@@ -6,50 +6,57 @@ title: HelperOptions
...
-        'type' => {'text', 'hidden', 'select', 'bool', 'radio',    // The kind of input field you want to use.
+$this->fields_options = [
+  'general' => [
+    'title' => $this->l('Carrier options'),                      // The title of the fieldset. If missing, default is 'Options'.
+    'top' => $this->l('Text to display before the fieldset'),    // This text is display right above the first. Rarely used.
+    'image' => 'url to icon',                                    // If missing, will use the default icon for the tab.
+    'description' => $this->l('Display as description'),         // Displays an informational box above the fields.
+    'info' => $this->l('Display as info'),                       // Displays an unstyled text above the fields.
+    'fields' => [                                           // The various option fields.
+      'PS_CARRIER_DEFAULT' => [                             // The array is named after the option's ID. It must be the
```
Comment (atomiix) [inline development/components/helpers/helperoptions.md]:
```suggestion
    'info' => $this->l('Display as info'),                       // Displays an unstyled text above the fields.
    'fields' => [                                                // The various option fields.
      'PS_CARRIER_DEFAULT' => [                                  // The array is named after the option's ID. It must be the
```

**Review by ghost — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
thank you @jf-viguier  !

**Comment by eternoendless** [comment]
This one needs to be backported to 8.x

**Comment by jf-viguier** [comment]
@eternoendless done here https://github.com/PrestaShop/docs/pull/1212

**Pull request merged**